### ISSUE TITLE
Unable to delete fields from Content Types 

### ISSUE THREAD

**Issue by bryanboza** [dotCMS/core#10648]
We are having problems to delete some field from a new content  type 

## Expected Behavior
We should be able to add//remove multiple fields from the content types without problems 

## Current Behavior
We are showing an error and the field is not deleted 

```[06/02/17 11:51:30:280 CST]  INFO job.DeleteFieldJob: triggerJobImmediately : Deleting Field '123' for Structure with id: b2773792-ef79-4bf3-8ba2-f236c5e0de5c
[06/02/17 11:51:30:286 CST] ERROR job.DeleteFieldJob: Unable to delete field '123'. Field Inode: 44f981e9-db80-4c84-8361-058fe2abf4da, Structure Inode: b2773792-ef79-4bf3-8ba2-f236c5e0de5c
com.dotmarketing.exception.DotDataException: Error Clearing Field '123' for Structure with id: b2773792-ef79-4bf3-8ba2-f236c5e0de5c
	at com.dotcms.content.elasticsearch.business.ESContentFactoryImpl.clearField(ESContentFactoryImpl.java:2293) ~[dotcms_4.0_5dc52df.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.cleanField(ESContentletAPIImpl.java:826) ~[dotcms_4.0_5dc52df.jar:?]
	at com.dotmarketing.portlets.contentlet.business.ContentletAPIInterceptor.cleanField(ContentletAPIInterceptor.java:402) ~[dotcms_4.0_5dc52df.jar:?]
	at com.dotmarketing.quartz.job.DeleteFieldJob.execute(DeleteFieldJob.java:135) [dotcms_4.0_5dc52df.jar:?]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:223) [dot.quartz-all-1.8.6_2.jar:?]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549) [dot.quartz-all-1.8.6_2.jar:?]
Caused by: org.h2.jdbc.JdbcSQLException: Columna "text2" no encontrada
Column "text2" not found; SQL statement:
UPDATE contentlet SET "text2" = '' WHERE inode = ? [42122-176]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:344) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.message.DbException.get(DbException.java:178) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.message.DbException.get(DbException.java:154) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.table.Table.getColumn(Table.java:637) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.command.Parser.readTableColumn(Parser.java:700) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.command.Parser.parseUpdate(Parser.java:734) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.command.Parser.parsePrepared(Parser.java:456) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.command.Parser.parse(Parser.java:306) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.command.Parser.parse(Parser.java:278) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.command.Parser.prepareCommand(Parser.java:243) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.engine.Session.prepareLocal(Session.java:442) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.engine.Session.prepareCommand(Session.java:384) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1188) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:73) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.jdbc.JdbcCallableStatement.<init>(JdbcCallableStatement.java:53) ~[h2-1.3.176.jar:1.3.176]
	at org.h2.jdbc.JdbcConnection.prepareCall(JdbcConnection.java:898) ~[h2-1.3.176.jar:1.3.176]
	at org.apache.tomcat.dbcp.dbcp2.DelegatingConnection.prepareCall(DelegatingConnection.java:319) ~[tomcat-dbcp.jar:8.0.18]
	at org.apache.tomcat.dbcp.dbcp2.DelegatingConnection.prepareCall(DelegatingConnection.java:319) ~[tomcat-dbcp.jar:8.0.18]
	at com.dotcms.content.elasticsearch.business.ESContentFactoryImpl.clearField(ESContentFactoryImpl.java:2276) ~[dotcms_4.0_5dc52df.jar:?]
	... 5 more
[06/02/17 11:51:31:435 CST]  INFO ajax.PermissionAjax: ##=> permissionMap: {inode=48190c8c-42c4-46af-8d1a-0cd5db894797, roleId=654b0931-1027-41f7-ad4d-173115ed8ec1, permission=3, id=12224, type=individual, isBitPermission=true}
```

## Steps to Reproduce (for bugs)
1. Login as admin
2. Create a new content type 
3. Add a new field (text type)
4. Try to delete the field 

## Your Environment
Tested in the last master-4.0 // H2 // Chrome

**Comment by jgambarios** [comment]
I was unable to reproduce with postgres but I confirm it is failing with H2

**Comment by jgambarios** [comment]
@freddyucv please make sure to tests all the dbs before to mark this ticket as resolve and also we need functional tests, with the proper tests we could catch these errors faster.

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/10662

**Comment by bryanboza** [comment]
Fixed, tested on master-4.0 // h2 // FF

**Comment by bryanboza** [comment]
Still having problems... Now in Oracle https://gist.github.com/bryanboza/32652de058e1616828d22e3b9d8091ef

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/10785

**Comment by bryanboza** [comment]
New problem detected with the date fields on MSSQL https://cb.dotcms.com/job/Git_Master_Int_Funct_Test_Runner_4.0_MSSQL/164/testReport/

**Comment by jcastro-dotcms** [comment]
PRs:

* https://github.com/dotCMS/core/pull/10926
* https://github.com/dotCMS/core/pull/10928

**Comment by bryanboza** [comment]
Still having some problems with some fields types: 

- [ ] Deleting site or folder field https://gist.github.com/bryanboza/10c2818630bd43c70d9645dbc4ffd2e0

- [ ] Deletig the tag fields: https://gist.github.com/bryanboza/d83380d7c7ae07d12114e3d8a23c604b

**Comment by jgambarios** [comment]
Related and we need to fix also: https://github.com/dotCMS/core/issues/10756

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/11013

**Comment by bryanboza** [comment]
Fixed, tested on master-4.0 // H2 // FF
